nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -159,7 +167,7 @@ let
                   tries=3
                   success=false
                   while [[ $success != true ]] && [[ $tries -gt 0 ]]; do
-                    ${systemd}/bin/systemd-ask-password "Enter key for $ds:" | ${cfgZfs.package}/sbin/zfs load-key "$ds" \
+                    ${systemd}/bin/systemd-ask-password --timeout=0 "Enter key for $ds:" | ${cfgZfs.package}/sbin/zfs load-key "$ds" \
```

Comment:
This timeout should probably be configurable with an option like `boot.zfs.passwordTimeout` or something (and for that matter, so should the import timeout).

Reply:
Done. I still left the timeout to be infinite since I don't know why somebody want to have this time out by default.

> (and for that matter, so should the import timeout).

Which import timeout? You mean all the hardcoded retry amounts?

Reply:
@ReneHollander Yea, the amount of time it will loop trying to import the pool. Probably can be a separate PR though.